pytorch/src/main/java/org/bytedeco/pytorch/presets/torch.java

```diff
@@ -1522,235 +1203,1643 @@
+                // torch/csrc/jit/serialization/storage_context.h
+                // torch/csrc/jit/serialization/import.h
+                "caffe2/serialize/inline_container.h",
+                "caffe2/serialize/istream_adapter.h",
+                "caffe2/serialize/read_adapter_interface.h",
+                "caffe2/serialize/versions.h",
                 "torch/csrc/jit/serialization/unpickler.h",
-                "torch/csrc/jit/serialization/import.h",
-                "torch/csrc/jit/serialization/pickle.h",
-                "torch/csrc/jit/serialization/python_print.h",
-                "torch/csrc/jit/serialization/type_name_uniquer.h",
+                "torch/csrc/jit/frontend/script_type_parser.h",
+                "torch/csrc/jit/frontend/resolver.h",
+                "torch/csrc/jit/frontend/sugared_value.h",
+                "torch/csrc/jit/frontend/error_report.h",
+                "torch/csrc/jit/frontend/tree.h",
+                "torch/csrc/jit/frontend/lexer.h",
+                "torch/csrc/jit/frontend/parser_constants.h",
+                "torch/csrc/jit/frontend/strtod.h",
+                "torch/csrc/jit/frontend/schema_matching.h",
+                "torch/csrc/jit/frontend/versioned_symbols.h",
+                "torch/csrc/jit/frontend/tree_views.h",
+                "torch/csrc/jit/serialization/pickler.h",
+
+                // Parsed and for inclusion in JNI
+                // See also https://github.com/pytorch/pytorch/blob/main/docs/cpp/source/Doxyfile
+                // for an approximation of what should be in API in addition to torch.h
+                "torch/torch.h",
+                "ATen/native/TensorShape.h",
+                "torch/csrc/jit/runtime/custom_operator.h",
                 "torch/csrc/jit/serialization/storage_context.h",
-                "torch/csrc/jit/serialization/export.h",
+                "torch/csrc/jit/serialization/import.h",
 
-                "torch/arg.h",
-                "torch/enum.h",
-                "torch/types.h",
-                "torch/utils.h",
+                // For inclusion in JNI only (skipped in infoMap)
+                "torch/csrc/jit/runtime/instruction.h",
+                "torch/csrc/jit/serialization/source_range_serialization.h"
 
-                "torch/data.h",
-                "torch/data/example.h",
-                "torch/data/iterator.h",
-                "torch/data/worker_exception.h",
-                "torch/data/dataloader.h",
-                "torch/data/dataloader/base.h",
-                "torch/data/dataloader_options.h",
-                "torch/data/dataloader/stateful.h",
-                "torch/data/dataloader/stateless.h",
-                "torch/data/datasets.h",
-                "torch/data/datasets/base.h",
+            },
+            exclude = {
+                // We only want to include torch.h in JNI file and the
+                // additional .h listed above
+                // Listing all others for exclusion
```

Comment:
Why do you need to do that?

Reply:
I didn't find a way to specify distinct lists for parsing, and for inclusion from JNI. 
In JNI we only need `#include <torch/torch.h>` (and a small number of others that are not included automatically from torch.h). 
If we manually include the big list of parsed .h in addition of torch.h, we get some compilation errors like the one triggered by : [this check](https://github.com/pytorch/pytorch/blob/37359c36fdb413df3b02996eb0ea2433c147db34/c10/util/complex_math.h#L1-L4). Or errors related to order (the order needed for JavaCPP parsing might not be the same than the one for inclusion).

If you can think of a way to avoid this big copy/paste, that would be nicer. We could use templating from maven but I don't think it's worth. Maybe adding some JavaCPP property to allow explicit specification of the header list for JNI ?

Reply:
I know *some* files don't like to be reincluded, but it's a very small number of them. Why do you want to exclude all of them?

Reply:
We know that the JNI will compile correctly if we include the headers the way they are supposed to be included. So I preferred this systematic approach and reset the whole list instead of determining by try and error a minimal exclusion list that works. This was simpler at the cost of this copy/paste in the presets file.

Don't you think it could be a good idea to be able to decorrelate the two lists by adding a property like `jninclude` ?

Reply:
That sounds reasonable. It would default to the values of include/cinclude, but users could override it, sure.

Reply:
Do you mean the reordering of includes such as `Half.h` or `intrusive_ptr.h` ? Those were necessary for the parsing to work, so I don't think it's related. 
If we really want to stick to the order given by `g++ -M`, to ease maintenance, they are maybe some Info that could be added that would have the same effect that reordering.

Reply:
About non-important headers that were commented out manually because non parseable: I have some patch to the parser that would make them parse, but as this is not blocking, I'm not sure you want them as PR.

Reply:
> If we really want to stick to the order given by g++ -H, to ease maintenance, they are maybe some Info that could be added that would have the same effect as that reordering.

Right, ok... It'd be nice to automate this a bit more using clang from the LLVM presets, but anyway let's start by getting rid of the duplicate exclude list.

Reply:
Do you want me to propose a PR for this jniinclude property or are you doing it ?

Reply:
If you can yes please open a pull request. Thanks!